src/examples/libpmem2/map_multiple_files/map_multiple_files.c

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2020, Intel Corporation */
+
+/*
+ * map_multiple_files.c -- implementation of virtual address allocation example
+ */
+
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#ifndef _WIN32
+#include <unistd.h>
+#else
+#include <io.h>
+#endif
+#include <libpmem2.h>
+
+int
+main(int argc, char *argv[])
+{
+	char *addr;
+	int *fd;
+	int ret = 1;
+	struct pmem2_config *cfg;
+	struct pmem2_map **map;
+	struct pmem2_source **src;
+	struct pmem2_vm_reservation *rsv;
+	pmem2_memset_fn memset_fn;
+	size_t alignment;
+	size_t n_files = argc - 1;
+	size_t *file_size;
+	size_t offset = 0;
+	size_t reservation_size = 0;
+	int align_iter;
```

Comment:
you only need three of these `_iter` variables, the rest of them can be defined inside of the `for` loop.
We usually use `nalign` naming scheme for stuff like this.